File: Leetcode/unique_morse_code_word.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        set<string> ans;
        string morse[26] = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        for (int i = 0; i < words.size(); i++) {
            string s;
            for (int j = 0; j < words[i].size(); j++) {
                s += morse[words[i][j] - 'a'];
            }
            ans.insert(s);
        }
        return ans.size();
    }
};

File: Leetcode/Unique Morse Code word/unique_morse_code_word.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        String[] arr = new String[] {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        
        HashSet<String> set = new HashSet<>();
        for (String word : words) 
            set.add(transformMorse(word, arr));
        return set.size();
    }
    private String transformMorse(String word, String[] arr) {
        StringBuilder build = new StringBuilder();
        for (char c : word.toCharArray())
            build.append(arr[(int)c - 97]);
        return build.toString();
    }
}